packages/compass-indexes/src/modules/indexes.ts

```diff
@@ -198,12 +211,11 @@ const _convertToModels = (indexes: Document[]): IndexDefinition[] => {
   });
 };
 
-const _mapAndSort = (
-  indexes: Document[],
-  sortColumn: SortColumn,
-  sortOrder: SortDirection
-) => {
-  return _convertToModels(indexes).sort(
-    _getSortFunction(_mapColumnToProp(sortColumn), sortOrder)
-  );
+export const dropFailedIndex = (
+  id: string
+): ThunkAction<void, RootState, void, AnyAction> => {
+  return (dispatch) => {
+    dispatch(inProgressIndexRemoved(id));
+    dispatch(fetchIndexes());
```

Comment:
as we are just deleting the failed index from store, should we fetch indexes again?

Reply:
Why not if we rerender the table anyway? I think Maurizio wanted to run fetch by timer :) but we haven't come to an agreement on this. So I decided to use this event to actualize the list. We can also bring this at the design sync later.